source/initial-dns-seedlist-discovery/tests/README.rst

```diff
@@ -23,11 +23,13 @@ these tests::
   _mongodb._tcp.test3.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
   _mongodb._tcp.test5.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
   _mongodb._tcp.test6.test.build.10gen.cc.  86400  IN SRV  27017  localhost.build.10gen.cc.
+  _mongodb._tcp.test11.test.build.10gen.cc. 86400  IN SRV  27017  localhost.build.10gen.cc.
 
   Record                                    TTL    Class   Text
   test5.test.build.10gen.cc.                86400  IN TXT  "connectTimeoutMS=300000&socketTimeoutMS=300000"
   test6.test.build.10gen.cc.                86400  IN TXT  "connectTimeoutMS=200000"
   test6.test.build.10gen.cc.                86400  IN TXT  "socketTimeoutMS=200000"
+  test11.test.build.10gen.cc.               86400  IN TXT  "connectTime" "outMS=150000" "&socketT" "imeoutMS" "=" "250000"
```

Comment:
Order is guaranteed or not?

Reply:
Here's [the spec](https://tools.ietf.org/html/rfc1035#section-3.3.14):
```
3.3.14. TXT RDATA format

    +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
    /                   TXT-DATA                    /
    +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+

where:

TXT-DATA        One or more <character-string>s.

TXT RRs are used to hold descriptive text.  The semantics of the text
depends on the domain where it is found.
```

So it does not explicitly say that the order is guaranteed but I think it has to be in practice.

While looking this up I did find an interesting comment on [this mailing list](https://lists.isc.org/pipermail/bind-users/2009-March/075931.html):
```
Note that any DNS response which exceeds 512 bytes is slightly 
undesirable, since in the absence of EDNS0 (which the vast majority of 
-- but not all -- implementations honor these days), responses which 
exceed 512 bytes will signal truncation and prompt a retry via TCP. It's 
optimal to stay within 512 bytes if possible.
```

This might be because the spec limits UDP DNS messages to 512 bytes: https://tools.ietf.org/html/rfc1035#section-2.3.4

Reply:
From what I understood, the order of strings **in** an TXT record is ordered. There is just no guaranteed order among multiple TXT records.

The 512 byte limit is indeed because of UDP limitations. Above that, DNS is required to use TCP as per https://tools.ietf.org/html/rfc1035#section-4.2.1